idpf_specification.md

```diff
@@ -6433,28 +6433,101 @@ VIRTCHNL2_OP_RDMA			= 529,
 
-- Capability and Data structures
+### Capability and Data structures
   
-  The response from VIRTCHNL2_OP_GET_CAPS rrom CP will have the VIRTCHNL2_CAP_PTP information. Driver will use this info to either register the PTP interfaces to driver ro control the requests from the user/stack.
-
-- Configuration
+ This capability consists of multiple sub capabilities that can be exposed by a device and are divided primarily in a few categories as described below. The primary PTP capability request is made by IDPF driver through get_capabilities virtchannel command by setting VIRTCHNL2_CAP_PTP bit in the capability bitmap. If the device supports PTP capability the response from Device Control still has the VIRTCHNL2_CAP_PTP bit set in the capability bitmap.
+
+#### PTP Sub capabilities
+1.	Ability to synchronize system (OS) time with Device time by doing cross time stamping. 
+a.	In legacy devices with no PCIE PTM capability this was done by independently sampling device time and system time but doing it in fashion that they can be done as close in time as possible or by taking two system time sample one before taking the device time sample and one immediately after the device time sample and the mean system time value between the two would be approximately the same time when the device time was sampled.
+This capability is exposed as 
+VIRTCHNL2_PTP_CAP_GET_DEVICE_CLK_TIME
+VIRTCHNL2_PTP_CAP_GET_DEVICE_CLK_TIME_MB
+Note: The Mailbox (_MB) version to get Device time over mailbox is generally not in use but is for completeness as the Device time over mailbox cannot be relied as being done exactly in the middle of the two system time samples or due to proxy nature can be far off from the system time sample. The capability returns only one time value from the Device, that of the Device master time.
+b.	With PCIE PTM capability, the system OS time and the Device time can be latched simultaneously by the device. This is done by the Device’s ability to sample the system ART timer at the same time the device master timer is sampled. 
+This capability is exposed as
+VIRTCHNL2_PTP_CAP_GET_CROSS_TIME	
+VIRTCHNL2_PTP_CAP_GET_CROSS_TIME_MB 
+Either capability is as accurate as the other in sampling the system and device time as the Mailbox (_MB) mechanism also relies on the Device control plane to be able to latch both system time and the Device at the same instance. The mailbox method just allows for indirect access to reading the latched values by the device. This capability returns two time values one for the Device and one for the system (CPU complex)
+2.	Ability to do Tx time stamping of packets as they leave the Device (typically in the PHY).
```

Comment:
This is specifically about Tx at the PHY right?

Tx at the PKB is unconditionally supported.

Reply:
Correct.